tests/integ/scala/test_dataframe_suite.py

```diff
@@ -1813,9 +1809,10 @@ def test_createDataFrame_with_given_schema_timestamp(session):
     ]
     df = session.create_dataframe(data, schema)
     schema_str = str(df.schema)
+
     assert (
         schema_str
-        == "StructType([StructField('TIMESTAMP', TimestampType(tz=ntz), nullable=True), "
+        == f"StructType([StructField('TIMESTAMP', TimestampType({'' if local_testing_mode else 'tz=ntz'}), nullable=True), "
```

Comment:
if in live mode schema `TimestampTimeZone.NTZ`  is returned for input `StructField("timestamp", TimestampType()),`.

shouldn't we update our mocking implementation to align with the behavior?

Reply:
In some scenarios it returns as NTZ, but in others it just remains DEFAULT. I have not worked out the logic of when it favors one or the other. It does however always seem to expect the cast to act like NTZ. I expect there's some inconsistency with how we type TimestampType outside of local testing as well so I think resolving this will be a larger refactor than we want before local testing release.

Reply:
interesting, it's the scenario that the data contains timezone while the schema doesn't. but I agree this should not block our release